godot-core/src/private.rs

```diff
@@ -358,17 +367,20 @@ where
             // TODO write custom panic handler and move this there, before panic backtrace printing
             flush_stdout();
 
-            let guard = info.lock().unwrap();
-            let info = guard.as_ref().expect("no panic info available");
-
-            if print {
-                godot_error!(
-                    "Rust function panicked at {}:{}.\n  Context: {}",
-                    info.file,
-                    info.line,
-                    error_context()
-                );
-                //eprintln!("Backtrace:\n{}", info.backtrace);
+            // Handle panic info in debug mode
+            #[cfg(debug_assertions)]
+            {
+                if let Some(info) = info.lock().unwrap().as_ref() {
```

Comment:
Can you restore this to the previous statements, with separate `guard` variable and `expect` statement?

This PR should disable hooks in Release mode, but not otherwise change semantics.